RenderStateNotationParser/src/RenderStateNotationParserImpl.cpp

```diff
@@ -165,19 +164,56 @@ static void Deserialize(const nlohmann::json& Json, RayTracingPipelineNotation&
 namespace Diligent
 {
 
-IRenderStateNotationParserImpl::IRenderStateNotationParserImpl(IReferenceCounters* pRefCounters, const Char* StrData) :
+IRenderStateNotationParserImpl::IRenderStateNotationParserImpl(IReferenceCounters*                        pRefCounters,
+                                                               const RenderStateNotationParserCreateInfo& CreateInfo,
+                                                               IncludeHandlerRenderStateNotationParser&   IncludeHandler) :
     TBase{pRefCounters}
 {
-    m_pAllocator        = std::make_unique<DynamicLinearAllocator>(DefaultRawMemoryAllocator::GetAllocator());
-    nlohmann::json Json = nlohmann::json::parse(StrData);
+    VERIFY_EXPR(CreateInfo.StrData != nullptr || (CreateInfo.FilePath != nullptr && CreateInfo.pStreamFactory != nullptr));
 
-    for (auto const& Signature : Json["ResourceSignatures"])
+    m_pAllocator = std::make_unique<DynamicLinearAllocator>(DefaultRawMemoryAllocator::GetAllocator());
+
+    String Source;
+    if (CreateInfo.FilePath != nullptr && CreateInfo.pStreamFactory != nullptr)
     {
-        PipelineResourceSignatureDesc ResourceDesc = {};
-        Deserialize(Signature, ResourceDesc, *m_pAllocator);
-        VERIFY_EXPR(ResourceDesc.Name != nullptr);
-        m_ResourceSignatureNames.emplace(HashMapStringKey{ResourceDesc.Name, false}, static_cast<Uint32>(m_ResourceSignatures.size()));
-        m_ResourceSignatures.push_back(ResourceDesc);
+        VERIFY_EXPR(CreateInfo.StrData == nullptr);
+
+        RefCntAutoPtr<IFileStream> pFileStream;
+        CreateInfo.pStreamFactory->CreateInputStream(CreateInfo.FilePath, &pFileStream);
+
+        if (!pFileStream)
+            LOG_ERROR_AND_THROW("Failed to open file: '", CreateInfo.FilePath, "'.");
+
+        auto pFileData = DataBlobImpl::Create();
+        pFileStream->ReadBlob(pFileData);
+        Source.assign(reinterpret_cast<const char*>(pFileData->GetConstDataPtr()), pFileData->GetSize());
+    }
+    else
+    {
+        Source.assign(CreateInfo.StrData);
+    }
+
+    nlohmann::json Json = nlohmann::json::parse(Source);
+
+    for (auto const& Import : Json["Imports"])
```

Comment:
I actually don't like the fact that Parser contains a tree of include parsers inside it. This is not the best design.
A better approach is to recursively traverse all included files inside the constructor. In particular, if you move [this code](https://github.com/MikhailGorobets/DiligentTools/blob/parser_import/RenderStateNotationParser/src/RenderStateNotationParserImpl.cpp#L219-L298) into a separate function that takes json parser as parameter, you can simply recursively call this function. The function may then create parsers for its includes and call itself recursively. This way you will not need `MergeRenderStates` function at all.